sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-full-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,263 @@
+import { createFakeSINValidation } from "@sims/test-utils/factories/sin-validation";
+import { IsNull, Not, Repository } from "typeorm";
+import {
+  createTestingAppModule,
+  describeQueueProcessorRootTest,
+} from "../../../../../../test/helpers";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames } from "@sims/utilities";
+import { FullTimeECertProcessIntegrationScheduler } from "../ecert-full-time-process-integration.scheduler";
+import { createMock } from "@golevelup/ts-jest";
+import { Job } from "bull";
+
+describe(
+  describeQueueProcessorRootTest(QueueNames.FullTimeECertIntegration),
+  () => {
+    let app: INestApplication;
+    let processor: FullTimeECertProcessIntegrationScheduler;
+    let userRepo: Repository<User>;
+    let studentRepo: Repository<Student>;
+    let applicationRepo: Repository<Application>;
+    let studentAssessmentRepo: Repository<StudentAssessment>;
+    let educationProgramOfferingRepo: Repository<EducationProgramOffering>;
+    let msfaaNumberRepo: Repository<MSFAANumber>;
+    let disbursementOverawardRepo: Repository<DisbursementOveraward>;
+    let disbursementScheduleRepo: Repository<DisbursementSchedule>;
+    let disbursementValueRepo: Repository<DisbursementValue>;
+
+    beforeAll(async () => {
+      const { nestApplication, dataSource } = await createTestingAppModule();
+      app = nestApplication;
+      userRepo = dataSource.getRepository(User);
+      studentRepo = dataSource.getRepository(Student);
+      applicationRepo = dataSource.getRepository(Application);
+      studentAssessmentRepo = dataSource.getRepository(StudentAssessment);
+      educationProgramOfferingRepo = dataSource.getRepository(
+        EducationProgramOffering,
+      );
+      msfaaNumberRepo = dataSource.getRepository(MSFAANumber);
+      disbursementOverawardRepo = dataSource.getRepository(
+        DisbursementOveraward,
+      );
+      disbursementScheduleRepo = dataSource.getRepository(DisbursementSchedule);
+      disbursementValueRepo = dataSource.getRepository(DisbursementValue);
+      // Processor under test.
+      processor = app.get(FullTimeECertProcessIntegrationScheduler);
+    });
+
+    it("Should execute overawards deductions and calculate awards effective value", async () => {
+      // Arrange
+
+      // User
+      const savedUser = await userRepo.save(createFakeUser());
+      // Student.
+      const savedStudent = await studentRepo.save(createFakeStudent(savedUser));
+      // Student SIN Validation
+      savedStudent.sinValidation = createFakeSINValidation({
+        student: savedStudent,
+      });
+      await studentRepo.save(savedStudent);
+      // MSFAA Number.
+      const savedMSFAANumber = await msfaaNumberRepo.save(
+        createFakeMSFAANumber({ student: savedStudent }),
+      );
+      // Create and save application.
+      const fakeApplication = createFakeApplication({ student: savedStudent });
+      fakeApplication.applicationNumber = "ECERT_TEST";
```

Comment:
Why is it important to set the applicationNumber here and not use the random number?

Reply:
@andrepestana-aot as per PR description, e-cert is not in the scope of PR.

Reply:
I removed the line 95 as it was not important for the test and to avoid confusion in the future.